enzyme/Enzyme/MLIR/Passes/RemoveUnusedEnzymeOps.cpp

```diff
@@ -290,13 +291,25 @@ struct InitSimplify : public OpRewritePattern<enzyme::InitOp> {
   }
 };
 
+struct EnzymeOpsRemoverOpPattern
+    : public OpInterfaceRewritePattern<enzyme::EnzymeOpsRemoverOpInterface> {
+  using OpInterfaceRewritePattern<
+      enzyme::EnzymeOpsRemoverOpInterface>::OpInterfaceRewritePattern;
+
+  LogicalResult matchAndRewrite(enzyme::EnzymeOpsRemoverOpInterface iface,
+                                PatternRewriter &rewriter) const final {
+    return iface.removeEnzymeOps(rewriter);
```

Comment:
High level comment then we can merge: It's critical that we make sure that whatever region op we simplify doesn't have its interface called if there are interior ops for push/pop/set/get. Where is that check happening?

For example we could do something here that says 
```if (iface.walk(op)[]{ return isa enzyme push/pop op; })) return failure();
```

before calling iface.removeEnzymeOps

Reply:
So currently, implementers of the interface should make sure of that but that is not ideal.

I think in this case we want to process ops in post-order (like the walk did before) and newly created ops should be processed only if they are after in the post-order.

I will try to implement something like this.